webrender/src/device.rs

```diff
@@ -2078,24 +2160,44 @@ impl<B: hal::Backend> Device<B, hal::Graphics> {
     }*/
     fn generate_texture_id(&mut self) -> TextureId {
         let mut rng = rand::thread_rng();
-        let mut texture_id = 1; // 0 is used for invalid
+        let mut texture_id = INVALID_TEXTURE_ID + 1;
         while self.images.contains_key(&texture_id) {
-            texture_id = rng.gen_range::<u32>(1, u32::max_value());
+            texture_id = rng.gen_range::<u32>(INVALID_TEXTURE_ID + 1, u32::max_value());
         }
         texture_id
     }
 
+    fn generate_fbo_ids(&mut self, count: i32) -> Vec<FBOId> {
+        let mut rng = rand::thread_rng();
+        let mut fboids = vec!();
+        let mut fbo_id = DEFAULT_DRAW_FBO + 1;
+        for _ in 0..count {
+            while self.fbos.contains_key(&fbo_id) || fboids.contains(&fbo_id) {
+                fbo_id = rng.gen_range::<u32>(DEFAULT_DRAW_FBO + 1, u32::max_value());
```

Comment:
heh, that's tricky.
I think what we need is just a `Vec<Option<B::something>>` addressable by index, and so this generation routine would just pick the first unused index there, or allocate a new one